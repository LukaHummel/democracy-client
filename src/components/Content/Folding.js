import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import Ionicons from 'react-native-vector-icons/Ionicons';

const Wrapper = styled.View`
  margin-top: 11;
  background-color: #fff;
  border-radius: 5;
`;

const Header = styled.TouchableOpacity`
  flex-direction: row;
  padding-vertical: 9;
  padding-horizontal: 9;
`;

const Headline = styled.Text`
  flex: 1;
  font-size: 17;
`;

const CollapseIcon = styled(Ionicons).attrs({
  color: 'rgb(151, 151, 151)',
  name: 'ios-arrow-up-outline',
  size: 20,
})`
  align-self: flex-start;
  transform: ${({ open }) => (open ? 'rotate(0deg)' : 'rotate(180deg)')};
`;

const Divider = styled.View`
  height: 1;
  background-color: #efeff4;
  margin-top: 9;
`;

const Text = styled.Text`
  padding-vertical: 9;
  padding-horizontal: 9;
  color: #666666;
`;

class Folding extends Component {
  state = {
    open: false,
  };

  render() {
    const { title, text } = this.props;
    const { open } = this.state;
    return (
      <Wrapper>
        <Header onPress={() => this.setState({ open: !open })}>
          <Headline>{title}</Headline>
          <CollapseIcon open={open} />
        </Header>
        {open && (
          <Fragment>
            <Divider />
            <Text>{text}</Text>
          </Fragment>
        )}
        <Divider />
      </Wrapper>
    );
  }
}

Folding.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
};

export default Folding;
