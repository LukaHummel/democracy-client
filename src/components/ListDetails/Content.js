import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const Wrapper = styled.ScrollView`
  flex: 1;
`;

const Text = styled.Text``;

const Content = ({ description }) => (
  <Wrapper>
    <Text>{description}</Text>
  </Wrapper>
);

Content.defaultProps = {
  description: '',
};

Content.propTypes = {
  description: PropTypes.string,
};

export default Content;
