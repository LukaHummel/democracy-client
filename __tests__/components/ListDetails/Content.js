import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Content from '../../../src/components/ListDetails/Content';

describe('test components/List/Row', () => {
  it('render row', () => {
    const wrapper = renderer.create(<Content description="Test description" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
