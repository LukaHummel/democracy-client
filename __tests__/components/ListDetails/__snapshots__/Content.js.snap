// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      Test description
    </Text>
  </View>
</RCTScrollView>
`;
